pkg/segment/segment.go

```diff
@@ -139,13 +142,21 @@ func getUserIdentity(telemetryFilePath string) (string, error) {
 }
 
 // SetError sanitizes any PII(Personally Identifiable Information) from the error
-func SetError(err error) string {
-	// Sanitize user information
+func SetError(err error) (errString string) {
+	// sanitize user information
 	user1, err1 := user.Current()
 	if err1 != nil {
 		return errors.Wrapf(err1, err1.Error()).Error()
 	}
-	return strings.ReplaceAll(err.Error(), user1.Username, "XXXX")
+	errString = strings.ReplaceAll(err.Error(), user1.Username, Sanitizer)
+
+	// sanitize file path
+	for _, str := range strings.Split(errString, " ") {
```

Comment:
maybe consider a more strict check for paths using regex? https://stackoverflow.com/a/33021907

Reply:
Why do you think the regexp will be more strict? That regexp is horribly unreadable and no one will understand it after a while.
If it can be done easily like this without regexp I would prefer that. It makes it obvious what is happening here.